pyproject.toml

```diff
@@ -0,0 +1,86 @@
+[project]
+name = "api"
+version = "0.1.0"
+description = "Codecov's API layer"
+readme = "README.md"
+requires-python = ">=3.12"
+dependencies = [
```

Comment:
TODO - these are ported over to align with existing requirements.in. Go through and pin these to max versions in existing requirements.txt so not introducing any regressions by using an old version of deps